mittab/libs/data_export/tab_card.py

```diff
@@ -0,0 +1,350 @@
+        4: ("LF", "WF"),
+        5: ("AD", "AD"),
+        6: ("AW", "AW"),
+    }
+    return victor_map[victor_code][side]
+
+
+def get_dstats(round_obj, deb1, deb2, iron_man):
+    """Taken from original tab card function.
+    Logic can probably be simplified, but keeping to
+    prevent breaking
+    """
+    dstat1 = [
+        k for k in RoundStats.objects.filter(debater=deb1).filter(round=round_obj).all()
+    ]
+    dstat2 = []
+    if not iron_man:
+        dstat2 = [
+            k
+            for k in RoundStats.objects.filter(debater=deb2)
+            .filter(round=round_obj)
+            .all()
+        ]
+    blank_rs = RoundStats(debater=deb1, round=round_obj, speaks=0, ranks=0)
+    while len(dstat1) + len(dstat2) < 2:
+        # Something is wrong with our data, but we don't want to crash
+        dstat1.append(blank_rs)
+    if not dstat2 and not dstat1:
+        return None, None
+    if not dstat2:
+        dstat1, dstat2 = dstat1[0], dstat1[1]
+    elif not dstat1:
+        dstat1, dstat2 = dstat2[0], dstat2[1]
+    else:
+        dstat1, dstat2 = dstat1[0], dstat2[0]
+    return dstat1, dstat2
+
+
+def json_get_round(round_obj, team, deb1, deb2):
+    chair = round_obj.chair
+    json_round = {
+        "round_number": round_obj.round_number,
+        "round_id": round_obj.pk,
+        "side": GOV if round_obj.gov_team == team else OPP,
+        "result": get_victor_label(
+            round_obj.victor, GOV if round_obj.gov_team == team else OPP
+        ),
+        "chair": chair.name,
+        "wings": [judge.name for judge in round_obj.judges.all() if judge != chair],
+    }
+
+    opponent = round_obj.opp_team if round_obj.gov_team == team else round_obj.gov_team
+    if opponent:
+        opponent_debaters = list(opponent.debaters.all())
+        json_round["opponent"] = {
+            "name": opponent.get_or_create_team_code(),
+            "school": opponent.school.name,
+            "debater1": opponent_debaters[0].name if opponent_debaters else None,
+            "debater2": (
+                opponent_debaters[1].name if len(
+                    opponent_debaters) > 1 else None
+            ),
+        }
+
+    json_round["debater1"] = list(
+        (stat.speaks, stat.ranks)
+        for stat in RoundStats.objects.filter(debater=deb1, round=round_obj)
+    )
+    json_round["debater2"] = (
+        list(
+            (stat.speaks, stat.ranks)
+            for stat in RoundStats.objects.filter(debater=deb2, round=round_obj)
+        )
+        if deb2
+        else []
+    )
+
+    try:
+        bye_round = Bye.objects.get(bye_team=team).round_number
+        json_round[bye_round - 1][bye_round] = "BYE"
+    except Bye.DoesNotExist:
+        pass
+
+    return json_round
+
+
+class JSONDecimalEncoder(json.JSONEncoder):
+    def default(self, o):  # pylint: disable=E0202
+        if isinstance(o, Decimal):
+            return float(o)
+        return super().default(o)
+
+
+def get_all_json_data():
```

Comment:
I'm able to trigger 500s by running an export on a tournament with slightly-incorrect/incomplete data. E.g. a round which doesn't have a chair. I think we should be able to return `null` in JSON for those instances